libs/multicall/src/hooks/useMultipleContractSingleData.ts

```diff
@@ -37,9 +37,31 @@ export function useMultipleContractSingleData<T = Result>(
     })
   }, [addresses, callData])
 
+  /**
+   * ChainId must be in useSWR deps bellow.
```

Comment:
I spend a day on that problem!
If anyone can explain me how it works, I will pay 50 ETH (Sepolia) as a bounty :) 
`useSWR` must be triggered when `provider` changes, but for some reason it doesn't!

Reply:
I think this is the closest of documentation on this regard https://swr.vercel.app/docs/arguments

Checking their code:
- https://github.com/vercel/swr/blob/main/src/index/use-swr.ts#L118
- Serialization: https://github.com/vercel/swr/blob/main/src/_internal/utils/serialize.ts#L6
- StableHash: https://github.com/vercel/swr/blob/main/src/_internal/utils/hash.ts#L23

Caveat: checked their main, might be different from the version we have.